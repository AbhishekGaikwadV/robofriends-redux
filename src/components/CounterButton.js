import React, {useState} from 'react';

const CounterButton = () => {
    const [count, setCount] = useState(0);

    const onIncrease = e => {
        setCount( (state) => state + 1);
    };

    return (
        <div>
            <div>Count: {count}</div>
            <button onClick={onIncrease}>Increase count</button>
        </div>
    );
};

export default CounterButton;
