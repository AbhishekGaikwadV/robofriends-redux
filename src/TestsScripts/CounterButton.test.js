import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CounterButton from '../components/CounterButton';

// npm test -- --coverage


describe("CounterButton", ()=>{
    it("CounterButton SnapShot", () => {
        expect(shallow(<CounterButton />)).toMatchSnapshot();
    });

    it("CounterButton State", () => {
        const wrapperMount = mount(<CounterButton />);

        wrapperMount.find('button').simulate('click');
        expect(wrapperMount.props()).toEqual(1);


    });
} );