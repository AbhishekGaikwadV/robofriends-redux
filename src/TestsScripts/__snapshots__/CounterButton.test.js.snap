// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton CounterButton SnapShot 1`] = `ShallowWrapper {}`;
